AE-1343 Add tests for viesti service

muep opened the pull request.

muep committed:
AE-1343 Add tests for viesti service

MattiNieminen commented on etp-backend/src/test/clj/solita/etp/test_data/viesti.clj:
```diff
@@ -0,0 +1,6 @@
+  (:require [solita.etp.schema.viesti :as viesti-schema]
+            [solita.etp.test-data.generators :as generators]))
+
+(defn complete-ketju-add [ketju-add]
```
Tänne voisi tehdä samannimiset funkkarit kuin muuallekin, eli generate-add yms.

muep replied:
Minusta vaikutti selkeämmältä tehdä osittainen objekti ja täyttää schemasta loput, verrattuna että tekee ensin täyden satunnaisen ja sitten korvaa kenttiä. Mutta kyllä tyylin yhtenäistämiseksi voisi myös tehdä tuolleen kuten muualla.

muep replied:
Jätin koko namespacen pois, kun näytti ettei sinne juurikaan tule contenttia.

muep committed:
AE-1406 Always handle :include-kasitelty

The previous ketju search queries had a trailing OR branch that would
ignore the parameter, making it impossible to hide processed ketjus in
some combinations of parameters.

The queries are rephrased so that it is hopefully easy to see that the
three available constraints can be applied independently from each
other.

muep commented:
Teen tätä nyt semmoisella ajatuksella, että AE-1406 / #639 on ihan validi asia.

muep commented on etp-backend/src/main/clj/solita/etp/service/viesti.clj:
```diff
@@ -63,7 +63,7 @@
        (viesti-db/select-viestit db {:id viestiketju-id :reader-id (:id whoami)})))
 
-(defn find-kayttajat [db]
+(defn- find-kayttajat [db]
```
This is coming in from https://github.com/solita/etp-core/pull/640

muep commented on etp-backend/src/main/sql/solita/etp/db/viesti.sql:
```diff
@@ -7,10 +7,9 @@ select
-        viestiketju.kasittelija_id = :kasittelija-id or
-        ((viestiketju.kasittelija_id is not null) = :has-kasittelija))) or
-  (:kasittelija-id::int is null and :has-kasittelija::boolean is null)
+  ((not viestiketju.kasitelty) or :include-kasitelty) and
```
https://github.com/solita/etp-core/pull/639

muep commented on etp-backend/src/main/sql/solita/etp/db/viesti.sql:
```diff
@@ -33,10 +32,9 @@ limit :limit offset :offset;
-        viestiketju.kasittelija_id = :kasittelija-id or
-        ((viestiketju.kasittelija_id is not null) = :has-kasittelija))) or
-  (:kasittelija-id::int is null and :has-kasittelija::boolean is null);
+  ((not viestiketju.kasitelty) or :include-kasitelty) and
```
https://github.com/solita/etp-core/pull/639
